names/names.py

```diff
@@ -13,44 +13,45 @@
 names_2 = f.read().split("\n")  # List containing 10000 names
 f.close()
 
+# STRECH
 
-#runtime: 0.23628687858581543 seconds
-# O( (log n) ^ 2)
-bst = BinarySearchTree(names_2.pop())
-for name_2 in names_2:
-    bst.insert(name_2)
 
+# runtime: 0.061779022216796875 seconds
+def binary_search(arr, target):
+    if len(arr) == 0:
+        return -1 # empty array
+    low = 0
+    high = len(arr)-1
+    while low <= high:
+        middle = (high+low)//2
+        if arr[middle] < target:
+        # get RHS
+            low = middle+1
+        elif arr[middle] > target:
+        # get LHS
+            high = middle-1
+        else:
+            return True #middle
+    return False # not found
+
+names_2.sort()
 duplicates = []
 for name_1 in names_1:
-    if bst.contains(name_1): 
+    if binary_search(names_2, name_1):
```

Comment:
nice going. personally, I like this better than risking having an unbalanced BST.